test/hotspot/jtreg/runtime/os/TestTracePageSizes.java

```diff
@@ -0,0 +1,279 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test id=no-options
+ * @summary Run test with no arguments apart from the ones required by
+ *          the test.
+ * @requires os.family == "linux"
+ * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log TestTracePageSizes
+ */
+
+/*
+ * @test id=large-page-size-1g
+ * @summary Run test with 1g pages on x64. Excluding ZGC since it fail
+ *          initialization if no large pages are available on the system.
+ * @requires os.family == "linux"
+ * @requires os.arch=="amd64" | os.arch=="x86_64"
+ * @requires vm.gc != "Z"
+ * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log -XX:+UseLargePages -XX:LargePageSizeInBytes=1g TestTracePageSizes
+*/
+
+/*
+ * @test id=compiler-options
+ * @summary Run test without segmented code cache. Excluding ZGC since it
+ *          fail initialization if no large pages are available on the system.
+ * @requires os.family == "linux"
+ * @requires vm.gc != "Z"
+ * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log -XX:-SegmentedCodeCache TestTracePageSizes
+ * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log -XX:-SegmentedCodeCache -XX:+UseLargePages TestTracePageSizes
+ * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log -XX:-SegmentedCodeCache -XX:+UseTransparentHugePages TestTracePageSizes
+ */
+
+/*
+ * @test id=with-G1
+ * @summary Run tests with G1
+ * @requires os.family == "linux"
+ * @requires vm.gc.G1
+ * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log -XX:+UseG1GC TestTracePageSizes
+ * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log -XX:+UseG1GC -XX:+UseLargePages TestTracePageSizes
+ * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log -XX:+UseG1GC -XX:+UseTransparentHugePages TestTracePageSizes
+*/
+
+/*
+ * @test id=with-Parallel
+ * @summary Run tests with Parallel
+ * @requires os.family == "linux"
+ * @requires vm.gc.Parallel
+ * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log -XX:+UseParallelGC TestTracePageSizes
+ * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log -XX:+UseParallelGC -XX:+UseLargePages TestTracePageSizes
+ * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log -XX:+UseParallelGC -XX:+UseTransparentHugePages TestTracePageSizes
+*/
+
+/*
+ * @test id=with-Serial
+ * @summary Run tests with Serial
+ * @requires os.family == "linux"
+ * @requires vm.gc.Serial
+ * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log -XX:+UseSerialGC TestTracePageSizes
+ * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log -XX:+UseSerialGC -XX:+UseLargePages TestTracePageSizes
+ * @run main/othervm -XX:+AlwaysPreTouch -Xlog:pagesize:ps-%p.log -XX:+UseSerialGC -XX:+UseTransparentHugePages TestTracePageSizes
+*/
```

Comment:
I see you don't explicitly test UseSHM. Because the number of tests would be too large?

Reply:
I think this is fine for now; hopefully we'll get rid of UseSHM.